src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -3071,14 +3075,25 @@ public object VisitPipeline(PipelineAst pipelineAst)
                     input = inputTemp;
                 }
 
-                Expression invokePipe = Expression.Call(
-                    CachedReflectionInfo.PipelineOps_InvokePipeline,
-                    input.Cast(typeof(object)),
-                    firstCommandExpr != null ? ExpressionCache.FalseConstant : ExpressionCache.TrueConstant,
-                    Expression.NewArrayInit(typeof(CommandParameterInternal[]), pipelineExprs),
-                    Expression.Constant(pipeElementAsts),
-                    redirectionExpr,
-                    _functionContext);
+                Expression invokePipe;
+                if (pipelineAst.BackgroundProcess)
```

Comment:
I was thinking this would go at the top of this method to avoid needlessly compiling any of the body. That would also eliminate the need for some of the tests that were added above. Is that possible?